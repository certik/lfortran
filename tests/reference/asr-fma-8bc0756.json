{
    "basename": "asr-fma-8bc0756",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/fma.f90",
    "infile_hash": "59fa49468c6dcd7b7bfbb4bfab840e6ec357824e0618eed22308d5f1",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-fma-8bc0756.stdout",
    "stdout_hash": "0a8976e4f4706cc79ccef32f0199c17de8edaabdd20a8214a5035bde",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}