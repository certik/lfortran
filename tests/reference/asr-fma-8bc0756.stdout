(TranslationUnit (SymbolTable 1 {fma: (Program (SymbolTable 2 {abs: (ExternalSymbol 2 abs 4 abs lfortran_intrinsic_math [] abs Private), abs@dabs: (ExternalSymbol 2 abs@dabs 4 dabs lfortran_intrinsic_math [] dabs Private), abs@sabs: (ExternalSymbol 2 abs@sabs 4 sabs lfortran_intrinsic_math [] sabs Private), pi: (Variable 2 pi Local (RealConstant   3.14000000000000012e+00 (Real 4 [])) (RealConstant   3.14000000000000012e+00 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), s1: (Variable 2 s1 Local (Cast (RealConstant   1.00000000000000000e+00 (Real 4 [])) RealToReal (Real 8 []) (RealConstant   1.00000000000000000e+00 (Real 8 []))) () Save (Real 8 []) Source Public Required .false.), s2: (Variable 2 s2 Local (Cast (RealConstant   2.00000000000000000e+00 (Real 4 [])) RealToReal (Real 8 []) (RealConstant   2.00000000000000000e+00 (Real 8 []))) () Save (Real 8 []) Source Public Required .false.), sign: (ExternalSymbol 2 sign 4 sign lfortran_intrinsic_math [] sign Private), sign@signr32: (ExternalSymbol 2 sign@signr32 4 signr32 lfortran_intrinsic_math [] signr32 Private), x: (Variable 2 x Local (RealConstant   4.00000000000000000e+00 (Real 4 [])) () Save (Real 4 []) Source Public Required .false.), z: (Variable 2 z Local (Cast (RealConstant   1.00000000000000000e+00 (Real 4 [])) RealToReal (Real 8 []) (RealConstant   1.00000000000000000e+00 (Real 8 []))) () Save (Real 8 []) Source Public Required .false.)}) fma [] [(= (Var 2 x) (RealBinOp (RealBinOp (RealBinOp (Var 2 x) Mul (Cast (IntegerConstant 4 (Integer 4 [])) IntegerToReal (Real 4 []) (RealConstant   4.00000000000000000e+00 (Real 4 []))) (Real 4 []) ()) Add (RealBinOp (RealConstant   5.00000000000000000e-01 (Real 4 [])) Mul (FunctionCall 2 sign@signr32 2 sign [((RealConstant   1.00000000000000000e+00 (Real 4 []))) ((Var 2 x))] (Real 4 []) () ()) (Real 4 []) ()) (Real 4 []) ()) Sub (RealBinOp (Var 2 x) Mul (Var 2 pi) (Real 4 []) ()) (Real 4 []) ()) ()) (= (Var 2 z) (RealBinOp (Var 2 s1) Add (RealBinOp (Var 2 z) Mul (Var 2 s2) (Real 8 []) ()) (Real 8 []) ()) ()) (If (RealCompare (FunctionCall 2 abs@sabs 2 abs [((RealBinOp (Var 2 x) Sub (RealConstant   3.93999999999999995e+00 (Real 4 [])) (Real 4 []) ()))] (Real 4 []) () ()) Gt (RealConstant   9.99999999999999955e-07 (Real 4 [])) (Logical 4 []) ()) [(ErrorStop ())] []) (If (RealCompare (FunctionCall 2 abs@dabs 2 abs [((RealBinOp (Var 2 z) Sub (Cast (RealConstant   3.00000000000000000e+00 (Real 4 [])) RealToReal (Real 8 []) (RealConstant   3.00000000000000000e+00 (Real 8 []))) (Real 8 []) ()))] (Real 8 []) () ()) Gt (Cast (RealConstant   9.99999999999999955e-07 (Real 4 [])) RealToReal (Real 8 []) (RealConstant   9.99999999999999955e-07 (Real 8 []))) (Logical 4 []) ()) [(ErrorStop ())] [])]), iso_c_binding: (IntrinsicModule lfortran_intrinsic_iso_c_binding), iso_fortran_env: (IntrinsicModule lfortran_intrinsic_iso_fortran_env), lfortran_intrinsic_builtin: (IntrinsicModule lfortran_intrinsic_builtin), lfortran_intrinsic_math: (IntrinsicModule lfortran_intrinsic_math)}) [])
